types/mithril/index.d.ts

```diff
@@ -1,863 +1,268 @@
+		/** The data to be interpolated into the URL and serialized into the querystring. */
+		data?: any;
+		/** A constructor to be applied to each object in the response. */
+		type?: new (o: any) => any;
+		/** The name of the function that will be called as the callback. */
+		callbackName?: string;
+		/** The name of the querystring parameter name that specifies the callback name. */
+		callbackKey?: string;
+		/** If false, redraws mounted components upon completion of the request. If true, it does not. */
+		background?: boolean;
 	}
 
-	/**
-	* This is the base interface for property getter-setters
-	*
-	* @see m.prop
-	*/
-	interface Property<T> {
-		/**
-		* Gets the contained value.
-		*
-		* @return The contained value.
-		*/
-		(): T;
-
-		/**
-		* Sets the contained value.
-		*
-		* @param value The new value to set.
-		* @return The newly set value.
-		*/
-		(value: T): T;
+	export interface JsonpOptionsAll extends JsonpOptions {
+		/** The URL to send the request to. */
+		url: string;
 	}
 
-	/**
-	* This represents a non-promise getter-setter functions.
-	*
-	* @see m.prop which returns objects that implement this interface.
-	*/
-	interface BasicProperty<T> extends Property<T> {
-		/**
-		* Makes this serializable to JSON.
-		*/
-		toJSON(): T;
+	export interface Jsonp {
+		/** Makes a JSON-P request and returns a promise. */
+		<T>(options: JsonpOptionsAll): Promise<T>;
+		/** Makes a JSON-P request and returns a promise. */
+		<T>(url: string, options?: JsonpOptions): Promise<T>;
 	}
 
-	/**
-	* This represents a key-value mapping linking routes to components.
-	*/
-	interface Routes {
-		/**
-		* The key represents the route. The value represents the corresponding
-		* component.
-		*/
-		[key: string]: Component<Controller>;
+	export interface RequestService {
+		request: Request;
+		jsonp: Jsonp;
 	}
 
-	/**
-	* This represents a Mithril deferred object.
-	*/
-	interface Deferred<T> {
-		/**
-		* Resolve this deferred's promise with a value.
-		*
-		* @param value The value to resolve the promise with.
-		*/
-		resolve(value?: T): void;
-
-		/**
-		* Reject this deferred with an error.
-		*
-		* @param value The reason for rejecting the promise.
-		*/
-		reject(reason?: any): void;
-
-		/**
-		* The backing promise.
-		*
-		* @see Promise
-		*/
-		promise: Promise<T>;
+	/** Renders a vnode structure into a DOM element. */
+	export type Render = (el: Element, vnodes: Children) => void;
+
+	export interface RenderService {
+		render: Render
 	}
 
-	/**
-	* This represents a thennable success callback.
-	*/
-	interface SuccessCallback<T, U> {
-		(value: T): U | Thennable<U>;
+	/** Manually triggers a redraw of mounted components. */
+	export type Redraw = () => void;
+
+	export interface RedrawService {
+		redraw: Redraw
+		render: Render
 	}
 
-	/**
-	* This represents a thennable error callback.
-	*/
-	interface ErrorCallback<T> {
-		(value: Error): T | Thennable<T>;
+	export interface Static extends Hyperscript {
```

Comment:
@spacejack 

You can do it like [this](http://www.typescriptlang.org/play/index.html#src=declare%20function%20withAttr(name%3A%20string%2C%20callback%3A%20(value%3A%20any)%20%3D%3E%20any)%3A%20(e%3A%20%7B%20currentTarget%3A%20any%2C%20%5Bp%3A%20string%5D%3A%20any%20%7D)%20%3D%3E%20void%3B%0D%0Adeclare%20function%20withAttr%3CT%3E(name%3A%20string%2C%20callback%3A%20(this%3A%20T%2C%20value%3A%20any)%20%3D%3E%20any%2C%20thisArg%3A%20T)%3A%20(e%3A%20%7B%20currentTarget%3A%20any%2C%20%5Bp%3A%20string%5D%3A%20any%20%7D)%20%3D%3E%20void%3B%0D%0A%0D%0Ainterface%20Static%20%7B%0D%0A%20%20%20%20withAttr%3A%20typeof%20withAttr%3B%0D%0A%7D%0D%0A%0D%0Adeclare%20const%20m%3A%20Static%3B%0D%0A%0D%0Am.withAttr(%22value%22%2C%20function%20()%20%7B%0D%0A%20%20%20%20const%20n%3A%20number%20%3D%20this%0D%0A%7D%2C%201)), using `typeof` to get the type. And yes, it works as you'd expect.